### PR TITLE
Issue 856 : Enable multi params group for LRScheduler

### LINKED ISSUE #856: Enable multi params group for LRScheduler and create_lr_scheduler_with_warmup
## 🚀 Feature

It would be very helpful to enable support of optimizer's multiple params group for `LRScheduler` and `create_lr_scheduler_with_warmup` :
```python
import torch
from torch.optim.lr_scheduler import ExponentialLR

t1 = torch.tensor([0.0])
t2 = torch.tensor([0.0])

opt = torch.optim.SGD([
    {"params": [t1], "lr": 1.0, "weight_decay": 0.0},
    {"params": [t2], "lr": 1.0, "weight_decay": 0.0001},
])

torch_lr_scheduler = ExponentialLR(optimizer=opt, gamma=0.98)
lr_values = [None] * 100
scheduler = create_lr_scheduler_with_warmup(torch_lr_scheduler,
                                            warmup_start_value=0.0,
                                            warmup_end_value=0.1,
                                            warmup_duration=10,
                                            output_simulated_values=lr_values)

lr_values = np.array(lr_values)
# lr_values.shape = (100, 3) <=> event index, lr for group 1, lr for group 2
# Plot simulated values
plt.plot(lr_values[:, 0], lr_values[:, 1], label="learning rate g1")
plt.plot(lr_values[:, 0], lr_values[:, 2], label="learning rate g2")
```

### PR DESCRIPTION
**Pull request by sdesrozis**
Fixes #856 

Description:

`LRScheduler` can handle scheduler based on optimizer with one `param_groups`. This PR enables multi `param_groups` of optimizer.

The main modification is the return of the method `get_param` : `LRScheduler` returns a list of size `len(param_groups)`. Other handlers returns scalars.

Check list:
* [x] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by sdesrozis** [comment]
The API of `create_lr_scheduler_with_warmup` must be modified to handle list of parameters
```python
import torch
from torch.optim.lr_scheduler import ExponentialLR

t1 = torch.tensor([0.0])
t2 = torch.tensor([0.0])

opt = torch.optim.SGD([
    {"params": [t1], "lr": 1.0, "weight_decay": 0.0},
    {"params": [t2], "lr": 1.0, "weight_decay": 0.0001},
])

torch_lr_scheduler = ExponentialLR(optimizer=opt, gamma=0.98)
lr_values = [None] * 100
scheduler = create_lr_scheduler_with_warmup(torch_lr_scheduler,
                                            warmup_start_value=[0.0, 0.1],
                                            warmup_end_value=[0.1, 0.2],
                                            warmup_duration=10,
                                            output_simulated_values=lr_values)

lr_values = np.array(lr_values)
# lr_values.shape = (100, 3) <=> event index, lr for group 1, lr for group 2
# Plot simulated values
plt.plot(lr_values[:, 0], lr_values[:, 1], label="learning rate g1")
plt.plot(lr_values[:, 0], lr_values[:, 2], label="learning rate g2")
```

@vfdev-5 thoughts?

**Comment by vfdev-5** [comment]
@sdesrozis let's keep the API simple without complicating with lists. 
If user needs to create a specific lr scheduling for each group, he/she need or should create a specific torch lr scheduler + create_lr_scheduler_with_warmup for each group.

**Comment by sdesrozis** [comment]
> @sdesrozis let's keep the API simple without complicating with lists.
> If user needs to create a specific lr scheduling for each group, he/she need or should create a specific torch lr scheduler + create_lr_scheduler_with_warmup for each group.

Ok.

I have to fix `create_lr_scheduler_with_warmup` to handle multiple `PiecewiseLinear`. This is because last `milestones_values` is related to `lr` of `param_groups`.

**Comment by sdesrozis** [comment]
```python
import matplotlib.pyplot as plt
import numpy as np
import torch
from torch.optim.lr_scheduler import ExponentialLR

from ignite.contrib.handlers.param_scheduler import create_lr_scheduler_with_warmup


t1 = torch.tensor([0.0])
t2 = torch.tensor([0.0])
t3 = torch.tensor([0.0])

opt = torch.optim.SGD([
    {"params": [t1], "lr": 0.7, "weight_decay": 0.0},
    {"params": [t2], "lr": 0.2, "weight_decay": 0.0001},
    {"params": [t3], "lr": 0.4, "weight_decay": 0.01},
])

torch_lr_scheduler = ExponentialLR(optimizer=opt, gamma=0.98)
lr_values = [None] * 100
scheduler = create_lr_scheduler_with_warmup(torch_lr_scheduler,
                                            warmup_start_value=0.0,
                                            warmup_end_value=0.1,
                                            warmup_duration=10,
                                            output_simulated_values=lr_values)

lr_values = np.array(lr_values)
# lr_values.shape = (100, 3) <=> event index, lr for group 1, lr for group 2
# Plot simulated values
plt.plot(lr_values[:, 0], lr_values[:, 1], label="learning rate g1")
plt.plot(lr_values[:, 0], lr_values[:, 2], label="learning rate g2")
plt.plot(lr_values[:, 0], lr_values[:, 3], label="learning rate g3")
plt.show()
```
Plot

Figure_1

@vfdev-5 @erip what do you think ?

**Comment by vfdev-5** [comment]
@sdesrozis Very nice ! Does it work with `warmup_end_value=None` such that end value is the one of optimizer ?

**Comment by sdesrozis** [comment]
> @sdesrozis Very nice ! Does it work with `warmup_end_value=None` such that end value is the one of optimizer ?

I will check that 👍🏻

**Comment by sdesrozis** [comment]
Case `warmup_end_value=None` after small fix

Figure_1

@vfdev-5
